planner/core/find_best_task.go

```diff
@@ -619,6 +621,9 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 	if prop.TaskTp == property.CopDoubleReadTaskType {
 		return invalidTask, nil
 	}
+	if !prop.IsEmpty() && !candidate.isMatchProp {
```

Comment:
One more question: 
* The check use for what ?
* Why `convertToIndexScan` doesn't have this check?